### PR TITLE
Fix nightly runs

### PR DESCRIPTION
**Pull request by smickovskid**
- Fix nightly runs

### CODE REVIEW

File: .github/workflows/e2e_testnet_daily.yml [hunk 1]
```diff
@@ -56,37 +117,56 @@ jobs:
...
+    needs: [ get_solana_sha ]
     outputs:
       exists: ${{ steps.check-image.outputs.exists }}
     steps:
       - name: Check if image exists
         id: check-image
-        uses: smartcontractkit/chainlink-github-actions/docker/image-exists@ad22fbd6f4d108b82aaf49b527bcf40f32babea8 # v2.2.1
+        uses: smartcontractkit/chainlink-github-actions/docker/image-exists@e865e376b8c2d594028c8d645dd6c47169b72974 # v2.2.16
         with:
           repository: chainlink-solana-tests
```
Comment (krebernisak) [inline .github/workflows/e2e_testnet_daily.yml:128]: What is this repository ID?

File: .github/workflows/e2e_testnet_daily.yml [hunk 2]
```diff
@@ -34,20 +33,82 @@ concurrency:
...
+        uses: ./.github/actions/setup-go
+        with:
+          only-modules: "true"
+      - name: Get the sha from go mod
+        id: getshortsha
+        run: |
+          short_sha=$(git rev-parse --short HEAD)
+          echo "short sha is: ${short_sha}"
+          echo "short_sha=${short_sha}" >> "$GITHUB_OUTPUT"
+      - name: Checkout solana
```
Comment (krebernisak) [inline .github/workflows/e2e_testnet_daily.yml:86]: What is this checkout used for and required?

File: .github/workflows/e2e_testnet_daily.yml [hunk 3]
```diff
@@ -34,20 +33,82 @@ concurrency:
...
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.event.merge_group.head_sha }}
+      - name: Setup Go
+        uses: ./.github/actions/setup-go
+        with:
+          only-modules: "true"
+      - name: Get the sha from go mod
```
Comment (krebernisak) [inline .github/workflows/e2e_testnet_daily.yml:80]: What does this have anything to do with go.mod here?

**Review by krebernisak — DISMISSED** [review]

File: .github/workflows/e2e_testnet_daily.yml (hunk 1 above)
Comment (smickovskid) [reply .github/workflows/e2e_testnet_daily.yml:128]: Internal Chainlink action to check fi image has been build already on this commit. It is used everywhere in tests

File: .github/workflows/e2e_testnet_daily.yml (hunk 3 above)
Comment (smickovskid) [reply .github/workflows/e2e_testnet_daily.yml:80]: Fixed naming

**Review by davidcauchi — APPROVED** [review]

**Pull request merged**